backend/src/typeorm/repository/User.ts

```diff
@@ -22,17 +22,31 @@ export class UserRepository extends Repository<User> {
     return usersIndiced
   }
 
-  async findBySearchCriteria(searchCriteria: string): Promise<User[]> {
+  async findBySearchCriteriaPagedFiltered(
+    select: string[],
+    searchCriteria: string,
+    filterCriteria: ObjectLiteral[],
+    currentPage: number,
+    pageSize: number,
+  ): Promise<[User[], number]> {
     return await this.createQueryBuilder('user')
+      .select(select)
       .withDeleted()
       .where(
-        'user.firstName like :name or user.lastName like :lastName or user.email like :email',
-        {
-          name: `%${searchCriteria}%`,
-          lastName: `%${searchCriteria}%`,
-          email: `%${searchCriteria}%`,
-        },
+        new Brackets((qb) => {
+          qb.where(
+            'user.firstName like :name or user.lastName like :lastName or user.email like :email',
+            {
+              name: `%${searchCriteria}%`,
+              lastName: `%${searchCriteria}%`,
+              email: `%${searchCriteria}%`,
+            },
+          )
+        }),
```

Comment:
It would be cool if this could also be part of `filterCritera`, but I assume that that is a bit more complex due to the `OR` statement.

Reply:
No, because of the brackets.
The Parameter for .where() are Brackets or ObjectLiteral[] so I cannot put them both in one array.
```ts
where(where: Brackets | string | ((qb: this) => string) | ObjectLiteral | ObjectLiteral[], parameters?: ObjectLiteral): this;
```
I could make the searchCriteria parameter as Brackets instead of string.
So that this code 
```ts
        new Brackets((qb) => {
          qb.where(
            'user.firstName like :name or user.lastName like :lastName or user.email like :email',
            {
              name: `%${searchCriteria}%`,
              lastName: `%${searchCriteria}%`,
              email: `%${searchCriteria}%`,
            },
          )
        }),
``` 
move from the UserRepository into the resolver function.